fe/fe-core/src/main/java/com/starrocks/analysis/ColumnDef.java

```diff
@@ -95,12 +99,17 @@ public ColumnDef(String name, TypeDef typeDef) {
     }
 
     public ColumnDef(String name, TypeDef typeDef, boolean isKey, AggregateType aggregateType,
-                     boolean isAllowNull, DefaultValueDef defaultValueDef, String comment) {
+                     Boolean isAllowNull, DefaultValueDef defaultValueDef, String comment) {
         this.name = name;
         this.typeDef = typeDef;
         this.isKey = isKey;
         this.aggregateType = aggregateType;
-        this.isAllowNull = isAllowNull;
+        if (isAllowNull == null) {
+            this.isAllowNull = true;
+            this.isAllowNullImplicitly = true;
+        } else {
+            this.isAllowNull = isAllowNull;
```

Comment:
Better to explicitly set `isAllowNullImplicitly` to `false`

Reply:
ok, changed.